### PR TITLE
Align React imports

### PR DESCRIPTION
**Pull request by kachkaev**
## 🌟 What is the purpose of this PR?

The goal of this PR is to reduce the entropy in the codebase by aligning React import style. See similar work in https://github.com/blockprotocol/blockprotocol/pull/441.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202542409311090/1202621858687196/f) _(internal)_

## 🔍 What does this change?

- `FC`, `VFC`, `VoidFunctionComponent` → `FunctionComponent`
- `import React from "react"` & `import * as React from "react"` → `import { ... } from "react"`
- see PR comments

## 📜 Does this require a change to the docs?

No

## ⚠️ Known issues

- See PR comments
- https://github.com/prettier/prettier/issues/11923

## 🐾 Next steps

- Consider removing `FunctionComponent` (possibly via https://github.com/gndelia/codemod-replace-react-fc-typescript)
- Introduce `simple-import-sort` ESLint plugin, similar to how this was done in https://github.com/blockprotocol/blockprotocol/pull/297

## 🛡 What tests cover this?

CI

## ❓ How to test this?

Check CI and diff

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,6 +74,7 @@
     "concurrently": "7.0.0",
     "cross-env": "7.0.3",
     "dotenv-flow": "3.2.0",
+    "eslint-plugin-unicorn": "43.0.2",
```
Comment (kachkaev) [inline package.json:77]: We had this plugin mentioned in two workspaces, I moved it up. Eventually we want to keep it in`@hashintel/eslint-config` only, but this requires changes in half of the packages. ATM, only blocks extend their ESLint config from this package.

File: packages/blocks/image/src/components/resize-image-block.tsx [hunk 2]
```diff
@@ -1,4 +1,9 @@
-import React, { useLayoutEffect, useRef } from "react";
+import {
+  FunctionComponent,
+  MouseEvent as ReactMouseEvent,
```
Comment (kachkaev) [inline packages/blocks/image/src/components/resize-image-block.tsx:3]: Here is a slightly nasty part of using named React imports. `MouseEvent` from React shadows DOM `MouseEvent` and we can no longer use it. I had to import `MouseEvent as ReactMouseEvent` in four files. It was `React.MouseEvent` previously.

**Comment by nathggns** [comment]
Looking at this now

File: sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx [hunk 3]
```diff
@@ -10,7 +10,7 @@ date: "2022-06-14"
...
 
-It's now a lot easier to create blocks that are not tied to JavaScript, TypeScript, or React. As long as a block can use browser `Event`s, it can implement the Block Protocol. To show what the changes imply for adventurous block authors, this post will be a technical walk-through of how the Block Protocol can be implemented and used in a language that is not TypeScript. Specifically, we're going to build a block in F#—but the methodology and implementation details are applicable for other languages that support JavaScript as a transpilation target and JavaScript interoperability.
+It's now a lot easier to create blocks that are not tied to JavaScript, TypeScript, or As long as a block can use browser `Event`s, it can implement the Block Protocol. To show what the changes imply for adventurous block authors, this post will be a technical walk-through of how the Block Protocol can be implemented and used in a language that is not TypeScript. Specifically, we're going to build a block in F#—but the methodology and implementation details are applicable for other languages that support JavaScript as a transpilation target and JavaScript interoperability.
```
Comment (nathggns) [inline sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx]: This this is unintentional?

Comment (kachkaev) [reply sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx]: Hmm it's strange that this has shown up in a diff for you. I fixed this on Friday in https://github.com/hashintel/hash/pull/847/commits/6a6bd2328c9e6240d6c9be0162251adca2a3da5a. Can it be that you've been looking at the old version of this PR somehow?

File: packages/hash/frontend/src/pages/signup.page/org-create.tsx [hunk 4]
```diff
@@ -183,7 +189,7 @@ export const OrgCreate: VFC<OrgCreateProps> = ({
     <div className={tw`flex flex-col items-center`}>
       <h1 className={tw`text-3xl font-bold mb-12`}>Create a team workspace</h1>
       <div className={tw`text-center mb-6`}>
-        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+        {}
```
[the remark this answers is not in the document]
Comment (kachkaev) [reply packages/hash/frontend/src/pages/signup.page/org-create.tsx:192]: Good spot! We should add `--report-unused-disable-directives` to package.json ESLint scripts to clean up similar cases. VSCode removes them on save, but CI does not care.

[the remark this answers is not in the document]
Comment (kachkaev) [reply packages/hash/frontend/src/pages/signup.page/org-create.tsx]: Fixed in https://github.com/hashintel/hash/pull/847/commits/d357c6dfc06c3057307c0c8275821882c8c4a247

File: packages/blocks/image/src/components/resize-image-block.tsx (hunk 2 above)
Comment (kachkaev) [reply packages/blocks/image/src/components/resize-image-block.tsx]: Nice idea! Should be fixed in https://github.com/hashintel/hash/pull/847/commits/ae2aca943adb830e0182d5baf1ec8851b76fbbab

File: sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx (hunk 3 above)
Comment (nathggns) [reply sites/hashdev/src/_pages/blog/5_build-blocks-in-any-language.mdx]: Gotcha, yeah not sure how that happened

**Review by nathggns — APPROVED** [review]

**Pull request merged**